src/frontend/src/handler/create_source.rs

```diff
@@ -381,33 +383,43 @@ pub(crate) async fn bind_columns_from_source(
             )
         }
         (format @ (Format::Plain | Format::Upsert | Format::Debezium), Encode::Avro) => {
-            let (row_schema_location, use_schema_registry) =
-                get_schema_location(&mut format_encode_options_to_consume)?;
+            if format_encode_options_to_consume
+                .remove(AWS_GLUE_SCHEMA_ARN_KEY)
+                .is_none()
+            {
+                // Legacy logic that assumes either `schema.location` or confluent `schema.registry`.
+                // The handling of newly added aws glue is centralized in `connector::parser`.
+                // TODO(xiangjinwu): move these option parsing to `connector::parser` as well.
```

Comment:
Perhaps we need to add logic to check `AWS_GLUE_SCHEMA_ARN_KEY` is not specified together with other 2.

Reply:
Just tested and when they are specified together:
```
NOTICE:  Get unknown format_encode_options for Plain Avro: schema.location
```

So it is already covered by the *unknown options check*, just not fatal. After the config parsing refactor we can try to make unknown options a fatal error.